crates/revm/src/evm.rs

```diff
@@ -34,31 +34,30 @@ use revm_precompile::Precompiles;
 /// # struct SomeDatabase;  // Mocking a database type for the purpose of this example
 /// # struct Env;           // Assuming the type Env is defined somewhere
 ///
-/// let evm: EVM<SomeDatabase> = EVM::new();
-/// assert!(evm.db.is_none());
+/// let evm: EVM<SomeDatabase> = EVM::new(SomeDatabase);
 /// ```
 ///
 #[derive(Clone)]
 pub struct EVM<DB> {
     pub env: Env,
-    pub db: Option<DB>,
+    pub db: DB,
 }
 
-pub fn new<DB>() -> EVM<DB> {
-    EVM::new()
+pub fn new<DB>(db: DB) -> EVM<DB> {
```

Comment:
Would love to refresh this in the future, so to not break API two times can you revert this?